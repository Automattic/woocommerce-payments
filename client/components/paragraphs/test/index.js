/** @format */

/**
 * External dependencies
 */
import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import Paragraphs from '../';

describe( 'Paragraphs', () => {
	test( 'renders set of strings correctly.', () => {
		const strings = [ 'Paragraph 1', 'Paragraph 2', 'Paragraph 3' ];
		const { container } = render( <Paragraphs>{ strings }</Paragraphs> );
		expect( container ).toMatchSnapshot();
	} );

	test( 'renders empty set of strings correctly', () => {
		const strings = [];
		const { container } = render( <Paragraphs>{ strings }</Paragraphs> );
		expect( container ).toMatchSnapshot();
	} );

	test( 'renders missing strings correctly', () => {
		const { container } = render( <Paragraphs /> );
		expect( container ).toMatchSnapshot();
	} );
} );
