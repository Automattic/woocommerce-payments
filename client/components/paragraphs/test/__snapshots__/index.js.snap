// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraphs renders empty set of strings correctly 1`] = `null`;

exports[`Paragraphs renders missing strings correctly 1`] = `null`;

exports[`Paragraphs renders set of strings correctly. 1`] = `
Array [
  <p
    key="0"
  >
    Paragraph 1
  </p>,
  <p
    key="1"
  >
    Paragraph 2
  </p>,
  <p
    key="2"
  >
    Paragraph 3
  </p>,
]
`;
